src/server/pps/server/pipeline_controller.go

```diff
@@ -132,17 +140,8 @@ func (m *ppsMaster) newPipelineOp(ctx context.Context, pipeline string) (*pipeli
 
 	// add pipeline auth
 	pachClient := m.a.env.GetPachClient(ctx)
+	op.ctx = auth_middleware.ClearWhoAmI(pachClient.Ctx())
```

Comment:
What is going on here?

Reply:
This definitely deserves more of a comment, I'll put that in. We use the master context for pipeline operations but want it to be authorized as the pipeline itself. That's what the old code handles, but there's this extra layer where background tasks like the master don't have any auth of their own and might need to do operations requiring auth (but no permissions). To meet this need, they explicitly add the key WhoAmI uses to cache lookups and skip the token check. In order to switch to the pipeline auth, the old cached value should be removed.

Reply:
Moved it to a separate function which hopefully makes things clearer